src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -2671,7 +2677,9 @@ class RegisterHumongousWithInCSetFastTestClosure : public HeapRegionClosure {
           if (g1h->is_in_closed_subset(ct->addr_for(card_ptr))) {
             if (*card_ptr != G1CardTable::dirty_card_val()) {
               *card_ptr = G1CardTable::dirty_card_val();
-              _dcq.enqueue(card_ptr);
+              if (!G1BarrierSimple) {
+                _dcq.enqueue(card_ptr);
+              }
```

Comment:
original patch has
```

if (G1BarrierSkipDCQ) {
--
  | 2649 | assert(G1ConcRefinementThreads == 0, "Concurrent refine should be disabled");
  | 2650 | assert(G1ConcRSLogCacheSize == 0, "Concurrent refine should be disabled");
  | 2651 | } else {
  | 2652 | _dcq.enqueue(card_ptr);
  | 2653 | }


```

Reply:
I think it's ok to not have the assertion.